release-version-check/action.yml

```diff
@@ -0,0 +1,142 @@
+    required: false
+    default: ""
+  override-version:
+    description: "Override the current version"
+    required: false
+    default: ""
+outputs:
+  version:
+    description: "The new version being released"
+    value: ${{ steps.new-version.outputs.version }}
+  status:
+    description: "The status of the job as a whole: ('success', 'fail')"
+    value: ${{ steps.status.outputs.status}}
+runs:
+  using: "composite"
+  steps:
+    - name: Input validation
+      id: validation
+      shell: bash
+      run: |
+        release_valid_inputs=("Initial Release" "Redeploy" "Dry Run")
+        if [[ ! "${release_valid_inputs[@]}" =~ "${{ inputs.release-type }}" ]]; then
+          echo "[!] 'release-type' option only supports the following values: (${release_valid_inputs})"
+          echo "::set-output name=step-failed::true"
+        fi
+
+        project_type_valid_inputs=("ts" "csharp")
+        if [[ ! "${project_type_valid_inputs[@]}" =~ "${{ inputs.project-type }}" ]]; then
+          echo "[!] 'project-type' option only supports the following values: (${project_type_valid_inputs})"
+          echo "::set-output name=step-failed::true"
+        fi
+
+        if [ ! -f "${{ inputs.file }}" ]; then
+          echo "[!] The relative path '${{ inputs.file}}' doesn't exist"
+          echo "::set-output name=step-failed::true"
+        fi
+
+        monorepo_valid_inputs=("true" "false")
+        if [[ ! "${monorepo_valid_inputs[@]}" =~ "${{ inputs.monorepo }}" ]]; then
+          echo "[!] 'monorepo' option only supports the following values: (${monorepo_valid_inputs})"
+          echo "::set-output name=step-failed::true"
+        fi
+
+        if [ "${{ inputs.monorepo }}" == "true" ] && [ "${{ inputs.monorepo-project }}" == "" ]; then
+          echo "[!] using the 'monorepo' option requires the use of 'monorepo-project'"
+          echo "::set-output name=step-failed::true"
+        fi
+
+
+    - name: Get project new version
+      shell: bash
+      id: new-version
+      run: |
+        # Get version from file depending on project type
+        case "${{ inputs.project-type }}" in
+          "ts")
+            version=$( jq -r ".version" ${{ inputs.file }} )
+            ;;
+          "csharp")
+            version=$( grep -o "<Version>.*</Version>" ${{ inputs.file }} | grep -Eo "[0-9]{4}\.[0-9]{2}\.[0-9]+" )
+            ;;
+        esac
+        echo "::set-output name=version::$version"
+
+
+    - name: Get current released version
+      shell: bash
+      id: current-version
+      run: |
+        if [ "${{ inputs.override-version }}" == "" ]; then
+          # Pull the latest 
+          case "${{ inputs.monorepo }}" in
+            "true")
+              previous_release_tag_version=$(
+                curl -sL https://api.github.com/repos/$GITHUB_REPOSITORY/releases | \
+                jq -r 'first(.[] | select(.tag_name | startswith("${{ inputs.monorepo-project }}"))).tag_name'
```

Comment:
Should we make sure `inputs.monorepo-project` is all lowercase like the tags?

Reply:
Since the valid inputs for the `monorepo-project` variable is a very small set of strings, I decided to add in input validation instead of a more generic rule for parsing.